Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/distance.h

```diff
@@ -107,15 +113,11 @@ struct Distance_computation{
       if(d > hdist)
         hdist=d;
     }
-
-    // update max value stored in distance
-    double current_value = *distance;
-    while( current_value < hdist )
-    {
-      if(distance->compare_exchange_weak(current_value, hdist))
-        current_value = hdist;
-    }
+    if(hdist > distance)
+      distance = hdist;
   }
+
+  void join( Distance_computation& rhs ) {distance = std::max(rhs.distance, distance); }
```

Comment:
max is not protected, I guess the testsuite with the check is not using tbb

Reply:
Indeed.